### PR TITLE
Do not align not templated tests in suite with Test Template

### LINKED ISSUE #444: AlignTemplatedTestCases does not handle [Template] NONE as expected
If a test has set the template to none, I would not expect it to be formatted the same as the templates with a value. Even if the template has a value I would expect that the [] sections would not be aligned to the columns taking data. This is larger part of what is described in #443.

Example input:
```
Test Template       Test Commit Message

*** Test Cases ***    UNLINTED FILE    LINTED FILE
Misplaced Git Trailer    misplaced_trailer.txt    misplaced_trailer_linted.txt
Mixed Unordered List Symbols    mixed_unordered.txt    mixed_unordered_linted.txt
Incorrectly Ordered List    ordered.txt    ordered_linted.txt
Unaligned List Item Indention    unaligned.txt    unaligned_linted.txt
Garbage Commit Message    garbage.txt    garbage_linted.txt
Valid Commit Message    unchanged.txt    unchanged_linted.txt    ${type trailer}    ${type trailer}\n${trailer block}
Header Validation
    [Template]    NONE
    ${valid header} =    Validate Header    ${header}
    Should Be True    ${valid header}
    ${err invalid header} =    Validate Header    ${EMPTY}
    Should Not Be True    ${err invalid header}
Trailer Validation
    [Template]    NONE
    ${valid type trailer} =    Validate Type Trailer    ${type trailer}
    Should Be True    ${valid type trailer}
    ${err invalid trailer} =    Validate Type Trailer    ${EMPTY}
    Should Not Be True    ${err invalid trailer}
```

Generated output:
```
Test Template       Test Commit Message

*** Test Cases ***                  UNLINTED FILE               LINTED FILE
Misplaced Git Trailer               misplaced_trailer.txt       misplaced_trailer_linted.txt
Mixed Unordered List Symbols        mixed_unordered.txt         mixed_unordered_linted.txt
Incorrectly Ordered List            ordered.txt                 ordered_linted.txt
Unaligned List Item Indention       unaligned.txt               unaligned_linted.txt
Garbage Commit Message              garbage.txt                 garbage_linted.txt
Valid Commit Message                unchanged.txt               unchanged_linted.txt            ${type trailer}     ${type trailer}\n${trailer block}
Header Validation
                                    [Template]                  NONE
                                    ${valid header}=            Validate Header                 ${header}
                                    Should Be True              ${valid header}
                                    ${err invalid header}=      Validate Header                 ${EMPTY}
                                    Should Not Be True          ${err invalid header}
Trailer Validation
                                    [Template]                  NONE
                                    ${valid type trailer}=      Validate Type Trailer           ${type trailer}
                                    Should Be True              ${valid type trailer}
                                    ${err invalid trailer}=     Validate Type Trailer           ${EMPTY}
                                    Should Not Be True          ${err invalid trailer}
```

Expected output:
```
Test Template       Test Commit Message

*** Test Cases ***                  UNLINTED FILE               LINTED FILE
Misplaced Git Trailer               misplaced_trailer.txt       misplaced_trailer_linted.txt
Mixed Unordered List Symbols        mixed_unordered.txt         mixed_unordered_linted.txt
Incorrectly Ordered List            ordered.txt                 ordered_linted.txt
Unaligned List Item Indention       unaligned.txt               unaligned_linted.txt
Garbage Commit Message              garbage.txt                 garbage_linted.txt
Valid Commit Message                unchanged.txt               unchanged_linted.txt            ${type trailer}     ${type trailer}\n${trailer block}
Header Validation
    [Template]    NONE
    ${valid header} =    Validate Header    ${header}
    Should Be True    ${valid header}
    ${err invalid header} =    Validate Header    ${EMPTY}
    Should Not Be True    ${err invalid header}
Trailer Validation
    [Template]    NONE
    ${valid type trailer} =    Validate Type Trailer    ${type trailer}
    Should Be True    ${valid type trailer}
    ${err invalid trailer} =    Validate Type Trailer    ${EMPTY}
    Should Not Be True    ${err invalid trailer}                              
```

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #444

### CODE REVIEW

File: docs/releasenotes/4.0.0.rst [hunk 1]
```diff
@@ -307,3 +307,15 @@ Bugs
     # robotidy: off
     # robotidy: on
     *** Settings ***
+
+- non templated tests in suites with ``Test Template`` set will no longer be aligned together with templated tests by
```
Comment (mnojek) [inline docs/releasenotes/4.0.0.rst:311]: It's hard for me to understand this sentence :D 
I mean, it's correct, but just hard to get all the meaning out of it.

**Review by mnojek — APPROVED** [review]

File: docs/releasenotes/4.0.0.rst (hunk 1 above)
Comment (bhirsz) [reply docs/releasenotes/4.0.0.rst:311]: Yeah.. it's hard to write release notes. Maybe ``Non templated tests will no longer be aligned with templated tests by..`` ?

Comment (mnojek) [reply docs/releasenotes/4.0.0.rst:311]: Maybe `Test cases with no template will no longer be aligned together with other tests in a templated test case file.`?

Comment (bhirsz) [reply docs/releasenotes/4.0.0.rst:311]: I like it, I will go with it

**Pull request merged**